packages/frontend/src/components/Explorer/CustomDimensionModal/index.tsx

```diff
@@ -0,0 +1,204 @@
+import { BinType, fieldId, isCustomDimension } from '@lightdash/common';
+import {
+    Button,
+    Group,
+    Modal,
+    NumberInput,
+    Radio,
+    Stack,
+    TextInput,
+    Title,
+    Tooltip,
+} from '@mantine/core';
+import { useForm } from '@mantine/form';
+import { useEffect } from 'react';
+import useToaster from '../../../hooks/toaster/useToaster';
+import { useExplorerContext } from '../../../providers/ExplorerProvider';
+
+// TODO: edit custom dimension
+// TODO: preview custom dimension results
+
+export const CustomDimensionModal = () => {
+    const { showToastSuccess } = useToaster();
+    const { isOpen, isEditing, item } = useExplorerContext(
+        (context) => context.state.modals.customDimension,
+    );
+    const toggleModal = useExplorerContext(
+        (context) => context.actions.toggleCustomDimensionModal,
+    );
+
+    const customDimensions = useExplorerContext(
+        (context) =>
+            context.state.unsavedChartVersion.metricQuery.customDimensions,
+    );
+    const addCustomDimension = useExplorerContext(
+        (context) => context.actions.addCustomDimension,
+    );
+    const editCustomDimension = useExplorerContext(
+        (context) => context.actions.editCustomDimension,
+    );
+
+    const form = useForm({
+        initialValues: {
+            customDimensionLabel: '',
+            binType: BinType.FIXED_NUMBER,
+            binConfig: {
+                fixedNumber: {
+                    binNumber: 2,
+                },
+            },
+        },
+        validate: {
+            customDimensionLabel: (label) => {
+                if (!label) return null;
+
+                if (!item) return null;
+
+                if (isEditing && label === item.name) {
+                    return null;
+                }
+
+                return customDimensions?.some(
+                    (customDimension) => customDimension.name === label,
+                )
+                    ? 'Dimension with this label already exists'
+                    : null;
+            },
+        },
+    });
+
+    const { setFieldValue } = form;
+
+    useEffect(() => {
+        if (isEditing && isCustomDimension(item)) {
+            setFieldValue('customDimensionLabel', item.name);
+            setFieldValue('binType', item.binType);
+            setFieldValue(
+                'binConfig.fixedNumber.binNumber',
+                item.binNumber ? +item.binNumber : 0,
```

Comment:
should the default be 2 ?

Reply:
let's go with `1` since the backend allows that now?